pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go

```diff
@@ -273,33 +280,75 @@ func (r *GatewayInstanceReconciler) createOrUpdateDeployment(
 				secContext.Capabilities.Add = append(secContext.Capabilities.Add, "NET_BIND_SERVICE")
 			}
 
+			if fsGroup := gatewayInstance.Spec.PodTemplate.Spec.PodSecurityContext.FSGroup; fsGroup != 0 {
+				if securityContext == nil {
+					securityContext = &kube_core.PodSecurityContext{
+						FSGroup: &fsGroup,
+					}
+				} else {
+					securityContext.FSGroup = &fsGroup
+				}
```

Comment:
Maybe we could simplify it a little bit?
```suggestion
				if securityContext == nil {
					securityContext = &kube_core.PodSecurityContext{}
				}

				securityContext.FSGroup = &fsGroup
```